trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/VLM.py

```diff
@@ -140,131 +195,213 @@ def VLM(conditions,settings,geometry):
-    TLE = (XB1[LE_ind] - XA1[LE_ind])/ np.sqrt((ZB1[LE_ind]-ZA1[LE_ind])**2 + (YB1[LE_ind]-YA1[LE_ind])**2)
-    
+    TLE   = TAN_LE[:,LE_ind]
     B2_LE = B2[:,LE_ind]
-    T2    = np.broadcast_to(TLE*TLE,np.shape(B2_LE))
+    T2    = TLE*TLE
     STB   = np.zeros_like(B2_LE)
     STB[B2_LE<T2] = np.sqrt(T2[B2_LE<T2]-B2_LE[B2_LE<T2])
-
-    # Panel Dihedral Angle, using AH and BH location
-    D   = np.sqrt((YAH-YBH)**2+(ZAH-ZBH)**2)[LE_ind]
-
-    #SID = ((ZBH-ZAH)[LE_ind]/D) # Just the LE values
-    COD = ((YBH-YAH)[LE_ind]/D) # Just the LE values
+    
+    # DL IS THE DIHEDRAL ANGLE (WITH RESPECT TO THE X-Y PLANE) OF
+    # THE IR STREAMWISE STRIP OF HORSESHOE VORTICES. 
+    COD = np.cos(phi[0,LE_ind])  # Just the LE values
+    SID = np.sin(phi[0,LE_ind])  # Just the LE values
 
     # Now on to each strip
     PION = 2.0 /RNMAX
     ADC  = 0.5*PION
 
     # XLE = LOCATION OF FIRST VORTEX MIDPOINT IN FRACTION OF CHORD.
     XLE = 0.125 *PION
+    
+    GAF = 0.5 + 0.5 *RJTS**2
+
+    # CORMED IS LENGTH OF STRIP CENTERLINE BETWEEN LOAD POINT
+    # AND TRAILING EDGE? THIS PARAMETER IS USED IN THE COMPUTATION
+    # OF THE STRIP ROLLING COUPLE CONTRIBUTION DUE TO SIDESLIP.
+    X      = XCH                       #x-coord of load point (horseshoe centroid)
+    XTE    = (VD.XA_TE + VD.XB_TE)/2   #Trailing edge x-coord behind the control point?    
+    CORMED = XTE - X   
 
     # SINF REFERENCES THE LOAD CONTRIBUTION OF IRT-VORTEX TO THE
     # STRIP NOMINAL AREA, I.E., AREA OF STRIP ASSUMING CONSTANT
     # (CHORDWISE) HORSESHOE SPAN.    
-    SINF = ADC * DCP # The horshoe span lengths have been removed
+    SINF = ADC * DCP # The horshoe span lengths have been removed since VST/VSS == 1 always
 
     # Split into chordwise strengths and sum into strips    
-    CNC = np.add.reduceat(SINF,chord_breaks,axis=1)
+    # SICPLE = COUPLE (ABOUT STRIP CENTERLINE) DUE TO SIDESLIP.
+    CNC    = np.add.reduceat(SINF       ,chord_breaks,axis=1)
+    SICPLE = np.add.reduceat(SINF*CORMED,chord_breaks,axis=1)
 
     # COMPUTE SLOPE (TX) WITH RESPECT TO X-AXIS AT LOAD POINTS BY INTER
     # POLATING BETWEEN CONTROL POINTS AND TAKING INTO ACCOUNT THE LOCAL
     # INCIDENCE.    
     XX   = (RK - .75) *PION /2.0
-
-    X1c  = (XA1+XB1)/2
-    X2c  = (XA2+XB2)/2
-    Z1c  = (ZA1+ZB1)/2
-    Z2c  = (ZA2+ZB2)/2
-
-    SLOPE = (Z2c - Z1c)/(X2c - X1c)
     TX    = SLOPE - ZETA
     CAXL  = -SINF*TX/(1.0+TX**2) # These are the axial forces on each panel
     BMLE  = (XLE-XX)*SINF        # These are moment on each panel
 
     # Sum onto the panel
     CAXL = np.add.reduceat(CAXL,chord_breaks,axis=1)
     BMLE = np.add.reduceat(BMLE,chord_breaks,axis=1)
-       
+    
+    SICPLE *= (-1) * COSIN * COD * GAF
     DCP_LE = DCP[:,LE_ind]
+       
+    
+    # COMPUTE LEADING EDGE THRUST COEFF. (CSUC) BY CALCULATING
+    # THE TOTAL INDUCED FLOW AT THE LEADING EDGE. THIS COMPUTATION
+    # ONLY PERFORMED FOR COSINE CHORDWISE SPACING (LAX = 0).    
```

Comment:
Only perform with cosine in VORLAX or here as well?

Reply:
only perform COSINE in VORLAX. the function just below  this comment has another comment explaining that SUAVE will continue to do this calculation even with linear spacing until consine spacing is implemented

Reply:
put in an extra comment line directly in VLM explaining this